diff --git a/src/maildirimport.cpp b/src/maildirimport.cpp
index efdbf86b0f01f94e070800e2ff2e9089b18b1d1e..aa60fee559b3aef5ed91fae97f23f34804ce7a53 100644
--- a/src/maildirimport.cpp
+++ b/src/maildirimport.cpp
@@ -83,7 +83,11 @@ void MaildirImport::import(const QUrl &maildir, const QUrl &destinationDir)
                 msg->setContent(mailData);
                 msg->parse();
 
+#if KMIME_VERSION >= QT_VERSION_CHECK(6, 6, 44)
                 const auto subject = msg->subject(KMime::CreatePolicy::Create)->asUnicodeString();
+#else
+                const auto subject = msg->subject(true)->asUnicodeString();
+#endif
                 const auto content = msg->decodedBody();
 
                 QFile markdownFile(destinationDir.toLocalFile() + u'/' + cleanFileName(subject) + u".md"_s);
